cli/src/pcluster/models/cluster_slurm.py

```diff
@@ -104,3 +107,19 @@ def __init__(
     ):
         super().__init__(image, head_node, shared_storage, monitoring, tags, iam, custom_actions)
         self.scheduling = scheduling
+
+        for queue in self.scheduling.queues:
+            for compute_resource in queue.compute_resources:
+                self._add_validator(
+                    InstanceArchitectureCompatibilityValidator,
+                    instance_type=compute_resource.instance_type,
+                    architecture=self.architecture,
+                )
+                if compute_resource.efa:
```

Comment:
When seeing this conditional statement, it feels like the validator should be inside compute_resource.efa. And here comes to the discussion about if we should allow access parent information such as os and architecture inside efa section.
Anyway, I am fine with your current code.

Reply:
I agree, for now let's go ahead with this approach, that works as expected even if we're not sure it's the optimal one.